clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -10875,6 +11081,15 @@ static void ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D,
   case ParsedAttr::AT_SYCLIntelFPGAMaxConcurrency:
     handleSYCLIntelFPGAMaxConcurrencyAttr(S, D, AL);
     break;
+  case ParsedAttr::AT_SYCLAddIRAttributesFunction:
+    handleSYCLAddIRAttributesFunctionAttr(S, D, AL);
+    break;
+  case ParsedAttr::AT_SYCLAddIRAttributesKernelParameter:
+    handleSYCLAddIRAttributesKernelParameterAttr(S, D, AL);
+    break;
+  case ParsedAttr::AT_SYCLAddIRAttributesGlobalVariable:
+    handleSYCLAddIRAttributesGlobalVariableAttr(S, D, AL);
+    break;
```

Comment:
There's a fair amount of code duplication in all of these cases which lends a bit of credence to my worry that we really only need one attribute to accomplish this. I should note, we can give that one attribute as many public-facing spellings as we want (so we can keep the distinct spellings if that's of value to us, but still only have one semantic attribute in the AST).

Reply:
See https://github.com/intel/llvm/pull/5535#discussion_r817604902